docs/openapi.yaml

```diff
@@ -1,791 +1,1564 @@
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/404'
         '500':
-          description: Internal server error
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/500'
 
-    patch:
-      summary: Partially update a specific applied Plan
+  '/plan/search/{text}/{status}/{sort}/{order}/{cursorPrimary}/{cursorSecondary}':
+    get:
+      summary: Search for Plans, retrieving paginated results
       description: >
-        Performs a partial update to a specific applied Plan.
-      operationId: updatePartialPlanById
+        Returns a list of Plans with matching criteria and cursor pagination.
+      operationId: searchPlans
       tags:
         - Plan
       parameters:
-        - name: planId
-          in: path
-          description: ID of the plan to update
-          required: true
-          schema:
-            type: integer
-            format: int64
-          example: 72
-      requestBody:
-        required: true
-        description: A Plan object
-        content:
-          application/json:
-            schema:
-              $ref: '#/components/schemas/Plan'
+        - $ref: '#/components/parameters/Search_Text'
+        - $ref: '#/components/parameters/Plan_Search_GET_Status'
+        - $ref: '#/components/parameters/Plan_Search_GET_Sort'
+        - $ref: '#/components/parameters/Plan_Search_GET_Order'
+        - $ref: '#/components/parameters/Search_CursorPrimary'
+        - $ref: '#/components/parameters/Search_CursorSecondary'
       responses:
         '200':
-          description: Successfully updated Plan
-        '400':
-          description: Invalid request body
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/Plan_Search_GET_200'
+        '401':
+          $ref: '#/components/responses/401'
         '403':
-          description: Request not made by an authorized user
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/403'
         '404':
-          description: Plan with the specified `planId` not found
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/404'
         '500':
-          description: Internal server error
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/500'
 
-    delete:
-      summary: Remove a specific applied Plan
+  '/plan/{planId}/activity/{cursorPrimary}/{cursorSecondary}':
```

Comment:
Parameter examples seem to have some formatting issue.

![Capture2](https://user-images.githubusercontent.com/32185302/81131497-33336680-8f00-11ea-9676-c0ab50823aef.JPG)